app/code/community/HE/TwoFactorAuth/Model/Observer.php

```diff
@@ -255,6 +255,12 @@ public function callAdminAuthenticate($observer)
             } else {
                 Mage::logException($e);
             }
+            $adminSession = Mage::getSingleton('admin/session');
+            if ($adminSession->isLoggedIn()) {
+                // log out a locked user
+                $adminSession->unsetAll();
+                $adminSession->getCookie()->delete($adminSession->getSessionName());
+            }
```

Comment:
Looks good. I think we should separate our this into its own function so that we can re-use if needed.

```
public function logOutUser() {
    $adminSession = Mage::getSingleton('admin/session');
    if ($adminSession->isLoggedIn()) {
    // log out a locked user
    $adminSession->unsetAll();
    $adminSession->getCookie()->delete($adminSession->getSessionName());
}
```
Also does it logs in user after showing "**This account is locked.**" message?